controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -327,6 +339,17 @@ func (r *ROSAControlPlaneReconciler) reconcileDelete(ctx context.Context, rosaSc
 	}
 
 	if cluster.Status().State() != cmv1.ClusterStateUninstalling {
+		// Deleting MachinePools first.
+		deleted, err := r.isMachinePoolsDeleted(ctx, rosaScope)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if !deleted {
+			// Reconcile after 1 min giving time for machinePools to be deleted.
+			return ctrl.Result{RequeueAfter: time.Minute}, nil
+		}
```

Comment:
why do we call this code twice? we can just do it once at the start of `reconcileDelete()` at L303